guild.go

```diff
@@ -563,61 +425,35 @@ func (c *Client) CreateBan(guild, user objects.Snowflake, params *CreateGuildBan
 		return err
 	}
 
-	res, err := c.request(&request{
-		method:      http.MethodPut,
-		path:        fmt.Sprintf(GuildBanUserFmt, guild, user),
-		contentType: JsonContentType,
-		body:        data,
-	})
-	if err != nil {
-		return err
-	}
-
-	if err = res.ExpectsStatus(http.StatusNoContent); err != nil {
-		return err
-	}
-	return nil
+	return NewRequest().
+		Method(http.MethodPut).
+		Path(fmt.Sprintf(GuildBanUserFmt, guild, user)).
+		ContentType(JsonContentType).
+		Body(data).
+		Expect(http.StatusNoContent).
+		Send(c)
 }
 
 func (c *Client) RemoveGuildBan(guild, user objects.Snowflake, reason string) error {
-	res, err := c.request(&request{
-		method:      http.MethodDelete,
-		path:        fmt.Sprintf(GuildBanUserFmt, guild, user),
-		contentType: JsonContentType,
-		body:        nil,
-		reason:      reason,
-	})
-	if err != nil {
-		return err
-	}
-
-	if err = res.ExpectsStatus(http.StatusNoContent); err != nil {
-		return err
-	}
-
-	return nil
+	return NewRequest().
+		Method(http.MethodDelete).
+		Path(fmt.Sprintf(GuildBanUserFmt, guild, user)).
+		ContentType(JsonContentType).
+		Reason(reason).
+		Expect(http.StatusNoContent).
+		Send(c)
 }
 
 func (c *Client) GetGuildRoles(guild objects.Snowflake) ([]*objects.Role, error) {
-	res, err := c.request(&request{
-		method:      http.MethodGet,
-		path:        fmt.Sprintf(GuildRolesFmt, guild),
-		contentType: JsonContentType,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	if err = res.ExpectsStatus(http.StatusOK); err != nil {
-		return nil, err
-	}
-
-	var roles []*objects.Role
-	if err = res.JSON(&roles); err != nil {
-		return nil, err
-	}
-
-	return roles, nil
+	roles := []*objects.Role{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildRolesFmt, guild)).
+		ContentType(JsonContentType).
+		Expect(http.StatusOK).
+		Bind(roles).
```

Comment:
We need to pass a pointer to the slice here, otherwise it will automatically return an error.